sympy/physics/control/lti.py

```diff
@@ -0,0 +1,553 @@
+from sympy import Basic, Mul, degree, Symbol, expand, cancel, Expr
+from sympy.core.evalf import EvalfMixin
+from sympy.core.numbers import Integer
+from sympy.core.sympify import sympify, _sympify
+
+__all__ = ['TransferFunction',]
+
+
+class TransferFunction(Basic, EvalfMixin):
+    """TransferFunction(num, den, var)
+
+    A class for representing Transfer Functions. This class is used to represent
+    LTI (Linear, time-invariant) systems in transfer function form. The arguments
+    are ``num``, ``den``, and ``var``, where ``num`` and ``den`` are numerator and
+    denominator of the ``TransferFunction`` respectively, and the third argument is
+    a variable used to anchor the symbols. ``num`` and ``den`` can be either
+    sympy expressions or numbers, whereas ``var`` has to be a Symbol. This class is a
+    candidate for a full symbolic model rather than just being a simple solver.
+
+    Parameters
+    ==========
+
+    num : Numerator
+        Numerator of the Transfer Function.
+    den : Denominator
+        Denominator of the Transfer Function.
+    var : Variable
+        Variable used to anchor the symbols.
+
+    Raises
+    ======
+
+    TypeError
+        When `var` is not a Symbol or when incorrect type for `num` or `den`
+        is supplied.
+    ValueError
+        When `den` is zero.
+
+    Examples
+    ========
+
+    >>> from sympy.abc import s, p, a
+    >>> from sympy.physics.control.lti import TransferFunction
+    >>> tf1 = TransferFunction(s + a, s**2 + s + 1, s)
+    >>> tf1
+    TransferFunction(a + s, s**2 + s + 1, s)
+    >>> tf1.num
+    a + s
+    >>> tf1.den
+    s**2 + s + 1
+    >>> tf1.var
+    s
+
+    Using different polynomial variables.
+
+    >>> tf2 = TransferFunction(a*p**3 - a*p**2 + s*p, p + a**2, p)
+    >>> tf2
+    TransferFunction(a*p**3 - a*p**2 + p*s, a**2 + p, p)
+    >>> tf3 = TransferFunction((p + 3)*(p - 1), (p - 1)*(p + 5), p)
+    >>> tf3
+    TransferFunction((p - 1)*(p + 3), (p - 1)*(p + 5), p)
+
+    Using Float or Integer (or other constants) as numerator and denominator.
+
+    >>> tf4 = TransferFunction(1/2, 4, s)
+    >>> tf4.num
+    0.500000000000000
+    >>> tf4.den
+    4
+    >>> tf4.var
+    s
```

Comment:
Show examples using different operations: +, /, -, *, **, etc. and explain what they do.

Reply:
This is a new module and people will not be familiar with using it. You need to have some elaborate examples to show how it can be used.

Reply:
Also, since we don't have any separate methods for series and parallel connections and +,-,*, etc.  will be used for that, you should also explain which configurations are represented by different operations.